app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+
+  validates :email, presence: true, uniqueness: true
+  validates :email, format: {with: VALID_EMAIL_REGEX}
+  validates :email, length: {maximum: Settings.email.max_length}
+  validates :email, length: {minimum: Settings.email.min_length}
```

Comment:
recommend là viết gọn lại e nhé
```suggestion
  validates :email, presence: true, uniqueness: true,
   format: {with: VALID_EMAIL_REGEX},
   length: {maximum: Settings.email.max_length, minimum: Settings.email.min_length}
```